miner/src/pool/queue.rs

```diff
@@ -340,9 +349,17 @@ impl TransactionQueue {
 			current_id.checked_sub(gap)
 		};
 
-		let state_readiness = ready::State::new(client, stale_id, nonce_cap);
-
-		let removed = self.pool.write().cull(None, state_readiness);
+		let mut removed = 0;
+		let senders: Vec<Address> = {
```

Comment:
Shouldn't this be a set? Otherwise we may cull the same sender more than once.

Reply:
We need the senders to be ordered by priority (so that we cull best senders first - more probability that it will have any effect on the queue), `best_transactions` in the pool (That we take the senders from) contains exactly one transaction per sender.

Reply:
Is it worth adding a comment here to that effect?

Reply:
I didn't realize `best_transactions` only had the best tx from each sender 👍.